### PR TITLE
fix: fix Dockerfile when container and host are not same os and arch

### PR DESCRIPTION
**Pull request by fiam**
- Copy project before installing dependencies
- Delete node_modules before installing dependencies
- Use node ci --omit=dev instead of --only=production, the latter is
  deprecated

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,12 +11,14 @@ WORKDIR /usr/src/app
...
-# install dependencies
-RUN npm ci --only=production
-
 # add project artifacts to docker image
 ADD . .
 
+# clean and reinstall dependencies, otherwise we'll get
+# incorrect binaries if the container and host are not
+# same OS and arch
+RUN rm -fr node_modules && npm ci --omit=dev
```
Comment (jensneuse) [inline Dockerfile:20]: @StarpTech please verify

Comment (StarpTech) [reply Dockerfile:20]: We don't need to delete it here. We will never copy node_modules https://github.com/wundergraph/docker/blob/main/.dockerignore#L2

File: Dockerfile [hunk 2]
```diff
@@ -11,12 +11,14 @@ WORKDIR /usr/src/app
 # rebuild image when package.json or lock has changed
 COPY package*.json ./
 
-# install dependencies
-RUN npm ci --only=production
```
Comment (StarpTech) [inline Dockerfile:15]: Why did you move this? It will invalidate all layers after every project change. We only want to invalidate the layers when the package*.json has changed.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (fiam) [reply Dockerfile:20]: In my tests it does get copied anyway, using `Docker version 20.10.17, build 100c701`

File: Dockerfile (hunk 2 above)
Comment (fiam) [reply Dockerfile:15]: I did not remove it, it's run with the same command as `rm -fr node_modules` to avoid creating an extra layer. I changed `--only=production` to `--omit=dev` because the former is deprecated.

File: Dockerfile (hunk 1 above)
Comment (StarpTech) [reply Dockerfile:20]: It shouldn't. We use it in several other places in the cloud.

File: Dockerfile (hunk 2 above)
Comment (StarpTech) [reply Dockerfile:15]: I said "move" :smile:

**Comment by fiam** [comment]
I've just tested and `node_modules` is not copied. Not sure why it was ignored previously, but it seems there was due to a problem in my local setup. Only change required would be swapping `--only=production` to `--omit=dev`, but I think that's better on a separate PR.

**Comment by StarpTech** [comment]
@fiam thanks
